### PR TITLE
🌱 Making workflows not run on forks.

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
This PR is making sure that github workflows only run on upstream repo and not on forks.

### CODE REVIEW

**Comment by SchSeba** [comment]
Hi @adilGhaffarDev thanks for the PR!
question instead can you just disable the github actions in your fork?

**Comment by tuminoid** [comment]
> Hi @adilGhaffarDev thanks for the PR! question instead can you just disable the github actions in your fork?

My 2 cents: It makes no sense for fork to try push their images to upstream repository, so wouldn't it make more sense to disable that kind of workflow here than each individual fork doing it? This PR is not disabling build and test type workflows.

**Review by SchSeba — APPROVED** [review]

**Comment by tuminoid** [comment]
Can we get the stuck Mellanox action retriggered to get this merged? Its been pending a long while... @SchSeba

**Comment by SchSeba** [comment]
Hi @tuminoid that lane is not critical we are just waiting for another maintainer to approve so we can merge that is the policy

cc @Eoghan1232 @e0ne

**Comment by tuminoid** [comment]
> Hi @tuminoid that lane is not critical we are just waiting for another maintainer to approve so we can merge that is the policy
> 
> cc @Eoghan1232 @e0ne

Oh, ok. GH shows its ready to go, so I guess there is no actual branch protection configured, but just agreed policy? Thanks!

**Comment by SchSeba** [comment]
That is right

File: .github/workflows/image-push-master.yml
Comment (adrianchiris) [inline .github/workflows/image-push-master.yml:8]: nit: any chance to have this reference a variable ?

similar to[1]

[1] https://github.com/k8snetworkplumbingwg/multus-cni/blob/89188023ef7d07b9a5a77633259a870f772d54ed/.github/workflows/image-push-master.yml#L6

Comment (adilGhaffarDev) [reply .github/workflows/image-push-master.yml]: done

**Review by Eoghan1232 — APPROVED** [review]
lgtm - thanks!

**Comment by adilGhaffarDev** [comment]
@Eoghan1232 @SchSeba @adrianchiris can we merge this one?

**Review by Eoghan1232 — APPROVED** [review]

**Review by zeeke — APPROVED** [review]
LGTM
thanks!

**Pull request merged**
